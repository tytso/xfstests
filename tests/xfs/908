#! /bin/bash
# SPDX-License-Identifier: GPL-2.0-or-newer
# Copyright (c) 2019, Oracle and/or its affiliates.  All Rights Reserved.
#
# FS QA Test No. 908
#
# Check that we can upgrade a filesystem to support bigtime and that inode
# timestamps work properly after the upgrade.

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1    # failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

# real QA test starts here
_supported_fs xfs
_supported_os Linux
_require_xfs_mkfs_crc
_require_xfs_mkfs_bigtime
_require_xfs_scratch_bigtime

date --date='Jan 1 00:00:00 UTC 2040' > /dev/null 2>&1 || \
	_notrun "Userspace does not support dates past 2038."

rm -f $seqres.full

# Format V5 filesystem without bigtime support and populate it
_scratch_mkfs -m crc=1,bigtime=0 > $seqres.full
_scratch_xfs_db -c 'version' -c 'sb 0' -c 'p' >> $seqres.full
_scratch_mount >> $seqres.full

touch $SCRATCH_MNT/a
touch $SCRATCH_MNT/b
ls -la $SCRATCH_MNT/* >> $seqres.full

_scratch_unmount
_check_scratch_fs

# Now upgrade to bigtime support
_scratch_xfs_admin -O bigtime >> $seqres.full
_check_scratch_fs
_scratch_xfs_db -c 'version' -c 'sb 0' -c 'p' >> $seqres.full

# Mount again, look at our files
_scratch_mount >> $seqres.full
ls -la $SCRATCH_MNT/* >> $seqres.full

# Modify some timestamps
touch -d 'Feb 22 22:22:22 UTC 2222' $SCRATCH_MNT/b

_scratch_cycle_mount

# Did the timestamp survive?
ls -la $SCRATCH_MNT/* >> $seqres.full
TZ=UTC stat -c '%Y' $SCRATCH_MNT/b

# success, all done
echo Silence is golden.
status=0
exit
