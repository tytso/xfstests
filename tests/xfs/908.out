QA output created by 908
7956915742
Silence is golden.
